homeassistant/helpers/area_registry.py

```diff
@@ -133,14 +135,20 @@ def _data_to_save(self) -> dict:
 @bind_hass
 async def async_get_registry(hass: HomeAssistantType) -> AreaRegistry:
     """Return area registry instance."""
-    task = hass.data.get(DATA_REGISTRY)
+    reg = hass.data.get(DATA_REGISTRY)
+
+    if not reg:
+        evt = hass.data[DATA_REGISTRY] = Event()
```

Comment:
Why do we need to change to awaiting an event?

Reply:
Why use a task to simulate an event? This is what events are designed for.

If you fork off a task and store it in the data, then you have to await it every run though this function. If you just store the bare registry then there's a quick path after the first load. 

The event only actually gets used in testing anyhow.

Reply:
Keep in mind that `async` functions can only context switch on code lines that have an `await` keyword. It's not like threaded code where you can have a context switch happen anywhere. If you're used to threaded code, this looks like it will be subject to a race condition, but every code path between awaits and the start/end of the function is executed atomically so there's no race condition.

Reply:
I think the old code is more clear. I'll let someone else comment further.

Reply:
I've rejigged the code a bit, hopefully that is more clear.

Reply:
This is an improvement, as it skips 1 yield to the event loop per fetching of the registry.